packages/contact-center/task/tests/helper.ts

```diff
@@ -0,0 +1,420 @@
+    expect(consoleErrorMock).not.toHaveBeenCalled();
+  });
+
+  it('should not call onAccepted or onDeclined if they are not provided', async () => {
+    const {result} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted: null, onDeclined: null, selectedLoginOption: 'BROWSER'})
+    );
+
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock);
+    });
+
+    act(() => {
+      result.current.accept();
+      result.current.decline();
+    });
+
+    await waitFor(() => {
+      expect(onAccepted).not.toHaveBeenCalled();
+      expect(onDeclined).not.toHaveBeenCalled();
+    });
+
+    // Ensure no errors are logged
+    expect(consoleErrorMock).not.toHaveBeenCalled();
+  });
+
+  it('should clean up task events on task change or unmount', async () => {
+    const {result, unmount} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock);
+    });
+
+    unmount();
+
+    await waitFor(() => {
+      expect(taskMock.off).toHaveBeenCalledWith(TASK_EVENTS.TASK_ASSIGNED, expect.any(Function));
+      expect(ccMock.off).toHaveBeenCalledWith(TASK_EVENTS.TASK_INCOMING, expect.any(Function));
+    });
+
+    // Ensure no errors are logged
+    expect(consoleErrorMock).not.toHaveBeenCalled();
+  });
+
+  it('should handle errors in accepting or declining tasks', async () => {
+    const failingTask = {
+      ...taskMock,
+      accept: jest.fn().mockRejectedValue('Error'),
+      decline: jest.fn().mockRejectedValue('Error'),
+    };
+
+    const {result} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    act(() => {
+      ccMock.on.mock.calls[0][1](failingTask);
+    });
+
+    act(() => {
+      result.current.accept();
+      result.current.decline();
+    });
+
+    await waitFor(() => {
+      expect(failingTask.accept).toHaveBeenCalled();
+      expect(failingTask.decline).toHaveBeenCalled();
+    });
+
+    // Ensure errors are logged in the console
+    expect(consoleErrorMock).toHaveBeenCalled();
+    expect(consoleErrorMock).toHaveBeenCalledWith('Error');
+  });
+});
+
+describe('useTaskList Hook', () => {
+  let consoleErrorMock;
+
+  beforeEach(() => {
+    // Mock console.error to spy on errors
+    consoleErrorMock = jest.spyOn(console, 'error').mockImplementation();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    consoleErrorMock.mockRestore();
+  });
+
+  it('should check onTaskAccepted and onTaskDeclined callbacks', async () => {
+    const {result} = renderHook(() => useTaskList({cc: ccMock, onTaskAccepted, onTaskDeclined}));
+
+    act(() => {
+      result.current.acceptTask(taskMock);
+      result.current.declineTask(taskMock);
+    });
+
+    await waitFor(() => {
+      expect(onTaskAccepted).toHaveBeenCalledWith(taskMock);
+      expect(onTaskDeclined).toHaveBeenCalledWith(taskMock);
```

Comment:
Also, did we try to fail these tests by either doing `not.toHaveBeenCalled` or by changing the `taskMock` value?